### PR TITLE
example-msg-generator: support fast message generation

### LINKED ISSUE #3241: example-msg-generator: support fast message generation
`example-msg-generator` generates one message per second by default. You can control the speed using the `freq` option. For example `freq(0.001)` generates one message in every 0.001 seconds.

On the other hand, it does not support busy message generation.

We use `example-msg-generator` in the test framework, for generating messages. In those cases, we just want to quickly generate messages, so that the tests quickly finish. As a workaround, we set some small value, that is accepted by grammar. For example:
https://github.com/syslog-ng/syslog-ng/blob/24f75c99e20156aa5d48bebf17801a50eb076651/tests/python_functional/functional_tests/template_functions/slog/test_secure_logging.py#L36

This is not ideal. 

This feature requests to accept `0` value in `freq`, which would make `example-msg-generator` generate messages without delay.

```
@version: 3.26

log {
   source { example-msg-generator(freq(0) num(100)); };
   destination { file("/dev/stdout"); };
};
```

Marking as good first issue.

One thing that worth attention: it is advisable to set a maximum generated message number with the `num` option, and/or set `flags(flow-control)` to be more gentle with the cpu resource.

### PR DESCRIPTION
**Pull request by vatsal-encs**
fixes #3241

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
"add to whitelist" add author of a Pull Request to whitelist (globally, be careful, it means this user can trigger kira for any PR)
do nothing -> CI won't start)

**Comment by Kokan** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
@vatsal-encs I mentioned a reference to an e2e test in the original issue. Could you update that too? From now `freq="0"` should work. It would implicitely test your code too.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: news/feature-3245.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+`example-msg-generator`: support fast message generation
```
Comment (furiel) [inline news/feature-3245.md:1]: This snippet will be added to the news file. We need something different than the original issue text.

```
`example-msg-generator`: support `freq(0)` for fast message generation

log {
   source { example-msg-generator(freq(0) num(100)); };
   destination { file("/dev/stdout"); };
};
```
is enough. The caution need to be added only to the documentation, if it is true at all. I have a little doubts now. I just tested your code, and did not run into trouble without flow control, so I may have been wrong about it.

Comment (vatsal-encs) [reply news/feature-3245.md]: @furiel : okay, thanks for the review

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**